File: framework/java/src/main/java/soda/web/http/ContentDisposition.java
package soda.web.http;

import soda.web.WebUtils;

public class ContentDisposition {

    public String name;

    public String filename;

    public boolean isFile() {
        return filename != null;
    }

    public static ContentDisposition fromHeaderValue(String value) {
        String name = WebUtils.findOne(value, "name=\"(.*?)\"").orElseThrow(() ->
            new IllegalArgumentException("no name found in Content-Disposition")
        );

        var cd = new ContentDisposition();
        cd.name = name;
        cd.filename = WebUtils.findOne(value, "filename=\"(.*?)\"").orElse(null);
        return cd;
    }

    public static ContentDisposition fromHeaderLine(String headerLine) {
        // no need to split by ':'
        return fromHeaderValue(headerLine);
    }

}

File: framework/scala/src/main/scala/soda/scala/web/http/ContentDisposition.scala
package soda.scala.web.http

import soda.scala.web.WebUtils

class ContentDisposition {

  var name: String = null

  var filename: String = null

  def isFile: Boolean = filename != null

}

object ContentDisposition {

  def fromHeaderValue(value: String): ContentDisposition = {
    WebUtils.findOne(value, "name=\"(.*?)\"") match {
      case Some(name) =>
        val cd = new ContentDisposition()
        cd.name = name
        cd.filename = WebUtils.findOne(value, "filename=\"(.*?)\"").orNull
        cd
      case None => throw new IllegalArgumentException("no name found in Content-Disposition")
    }
  }

  def fromHeaderLine(headerLine: String): ContentDisposition = {
    fromHeaderValue(headerLine)
  }

}
